crates/glue/cpp/src/internal.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+
+#define REFCOUNT_ONE ((ptrdiff_t)1 << (sizeof(ptrdiff_t) * 8 - 1))
+
+void *roc_alloc(size_t size, uint32_t alignment);
+void *roc_realloc(void *ptr, size_t new_size, size_t old_size, size_t alignment);
+void roc_dealloc(void *ptr, uint32_t alignment);
+void roc_panic(const char *message, uint32_t _tag_id);
```

Comment:
Just FYI this should take a Str in the first parameter

Reply:
Oh yes of course. We must have this bug in so many places. I copied this code from somewhere else in the repo.

Unfortunately, fixing this creates a cyclic dependency. I need to make `internal.h` depend on `str.h` but `str.h` already depends on `internal.h` for the allocation functions.

I might be able to break it by reorganising the files or pre-declaring something.